.github/workflows/publish.yml

```diff
@@ -148,6 +148,22 @@ jobs:
           overwrite: true
           tag: ${{ inputs.tag || 'nightly' }} # This will fail if `inputs.tag` is not a tag (e.g. testing a branch)
 
+      - name: Get formatted date
+        id: date
+        if: ${{ inputs.tag == '' && inputs.publish || github.event_name == 'schedule' }}
+        run: echo "date=$(date '+%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Upload binaries to nightly release with date tag
+        uses: svenstaro/upload-release-action@v2
+        if: ${{ inputs.tag == '' && inputs.publish || github.event_name == 'schedule' }}
+        with:
+          repo_name: noir-lang/noir
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ./nargo-${{ matrix.target }}.zip
+          asset_name: nargo-${{ matrix.target }}.zip
+          overwrite: true
+          tag: ${{ format('{0}-{1}', 'nightly', steps.date.outputs.date) }}
```

Comment:
I think this would no longer need nightly since we have date.

Reply:
I think we should call these "previews" maybe, so it'd be `preview-12-12-23` or whatever. I'm also fine keeping "nightly" but I want something in the tag to indicate these are not official releases

Reply:
Yeah something in the tag to indicate that its nightly makes sense. I think preview is clearer, though "nightly build" keeps us inline with Rust and is a common idiom, so would prefer to keep that terminology

Reply:
Hello @kevaundray 

Do we save the `nightly` prefix? Should I change it to `preview`?

Reply:
Hello, lets keep nightly prefix